### PR TITLE
fix: avoid all escaped characters replaced

### LINKED ISSUE #913: \@ in CSS Selector
* macOS Mojave 10.14 (18A391)
* Node.js 11.13.0
* npm 6.7.0
* webpack 4.29.6
* css-loader 2.1.1

### Expected Behavior

```HTML
<div class="m_x_@"></div>
```

### Actual Behavior

```SCSS
.m_x_\@ {
  margin-left: auto !important;
  margin-right: auto !important;
}
```

```JSX
/*
 * @flow
 */

import * as React from 'react';

import design from './design.scss';

class Test extends React.Component<P: {

}> {
  render () {
    return <div className={design['m_x_@']} />;
  }
}

export default Test;
```

```HTML
<div class="m_x_----wB9u"></div>
```

### Code

```js
const path = require('path');

const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const sass = require('sass');

const isProduction = process.env.NODE_ENV === 'production';

const js = {
  exclude: /node_modules/,
  test: /\.js$/,
  use: [
    {
      loader: 'babel-loader',
    },
  ],
};

module.exports = {
  devtool: isProduction ? '' : 'source-map',
  entry: {
    index: path.resolve(__dirname, 'private/index.js'),
  },
  mode: isProduction ? 'production' : 'development',
  module: {
    rules: [
      js,
      {
        exclude: /node_modules/,
        test: /\.scss$/,
        use: [
          MiniCssExtractPlugin.loader,
          {
            loader: 'css-loader',
            options: {
              importLoaders: 2,
              localIdentName: isProduction ? '[hash:base64:8]' : '[local]--[hash:base64:4]',
              modules: true,
              sourceMap: !isProduction,
            },
          },
          {
            loader: 'postcss-loader',
            options: {
              plugins: [
                require('autoprefixer'),
              ],
              sourceMap: !isProduction,
            },
          },
          {
            loader: 'sass-loader',
            options: {
              implementation: sass,
              outputStyle: isProduction ? 'compressed' : 'expanded',
              sourceMap: !isProduction,
            },
          },
        ],
      },
    ],
  },
  output: {
    filename: '[name].js',
    path: path.resolve(__dirname, 'public'),
  },
  plugins: [
    new MiniCssExtractPlugin(),
  ],
};
```

### PR DESCRIPTION
**Pull request by alreadyExisted**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

fixes #913 

### Breaking Changes

No

### Additional Info
No

### CODE REVIEW

**Comment by alreadyExisted** [comment]
> Good solution, but i thanks we should avoid this for all escaped characters, not only for `@`

Ok. I'll think about the solution and make changes.

**Comment by alreadyExisted** [comment]
Added solution for escaped characters, but i haven't won ascii hex escaped numeric `B\26 W\3F`. Too hard for me)

The code below does not work correctly.

```js
function normalizeIdentifier(value) {
  const escapedSymbolsMap = {
    '\\7E': '~',
    '': '!', 
    '\\40': '@', 
    '\\23': '#', 
    '\\24': '$',
    '\\25': '%', 
    '\\26': '&', 
    '\\5E': '^', 
    '': '*', 
    '': '(',
    '': ')', 
    '\\7B': '{', 
    '\\7D': '}', 
    '\\5B': '[', 
    '\\5D': ']',
    '\\60': '`', 
    '\\2F': '/', 
    '\\3D': '=', 
    '\\3F': '?', 
    '': '+',
    '\\5C': '\\', 
    '\\7C': '|', 
    '': '-', 
    '': '_', 
    '\\3A': ':', 
    '\\3B': ';',
    '': '\'', 
    '': '"', 
    '': ',', 
    '\\3C': '<',
    '': '.', 
    '\\5C': '>'
  };

  const escapedSymbols = Object.values(escapedSymbolsMap);
  const escapedSymbolsCode = Object.keys(escapedSymbolsMap).filter(item => item !== '');

  const identifiersRegExp = new RegExp(
    `[^a-zA-Z0-9${escapedSymbols.join('\\')}\\-_\u00A0-\uFFFF]`,
    'g'
  );

  escapedSymbolsCode.forEach(code => {
    const regExp = new RegExp(code, 'g');
    value = value.replace(regExp, escapedSymbolsMap[code]);
  });

  return value
    .replace(identifiersRegExp, '-')
    .replace(/^((-?[0-9])|--)/, '_$1');
}
```

File: test/fixtures/modules/localIdentName.css [hunk 1]
```diff
@@ -29,4 +29,9 @@
 :local(.m_x_\@) {
   margin-left: auto !important;
   margin-right: auto !important;
+}
+
+:local(.B\&W\?) {
+  margin-left: auto !important;
+  margin-right: auto !important;
 }
```
Comment (evilebottnawi) [inline test/fixtures/modules/localIdentName.css]: Add new line here

**Comment by alreadyExisted** [comment]
Done

**Comment by evilebottnawi** [comment]
Thanks for PR

**Pull request merged**